march_hardware/include/march_hardware/motor_controller/imotioncube/imotioncube_states.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Project March.
+#ifndef MARCH_HARDWARE_IMOTIONCUBE_STATES_H
+#define MARCH_HARDWARE_IMOTIONCUBE_STATES_H
+
+#include <string>
+#include "march_hardware/motor_controller/motor_controller_states.h"
+#include "imotioncube_state_of_operation.h"
+
+namespace march
+{
+struct IMotionCubeStates : public MotorControllerStates
+{
+public:
+  IMotionCubeStates() = default;
+
+  uint16_t statusWord;
+  std::string motionError;
+  std::string detailedError;
+  std::string secondDetailedError;
+  IMCStateOfOperation state;
+  std::string detailedErrorDescription;
+  std::string motionErrorDescription;
+  std::string secondDetailedErrorDescription;
+
+  virtual bool checkState()
```

Comment:
This method overrides the one from `MotorControllerStates`, therefore, it should add `override` and not `virtual`
```suggestion
  bool checkState() override
```